Assets/Scripts/Tools/EdgeTool.cs

```diff
@@ -0,0 +1,133 @@
+﻿using Assets.Scripts.Common.Extensions;
+using Assets.Scripts.Model;
+using UnityEngine;
+
+namespace Assets.Scripts.Tools
+{
+  internal enum EdgeMode { Create, Delete }
+
+  public class EdgeTool : ITool, IToolChangeObserver
+  {
+    private const float _nodeHitGlowStrength = 0.5f;
+    private readonly Color _createGlowColor = Color.green;
+    private readonly Color _deleteGlowColor = Color.red;
+    private readonly GraphService _graphService;
+    private EdgeMode _mode = EdgeMode.Create;
+    private Node _previouslyHitNode;
+
+    public EdgeTool(GraphService graphService)
+    {
+      _graphService = graphService;
+    }
+
+    private EdgeMode Mode
+    {
+      get => _mode;
+      set
+      {
+        if (_mode != value)
+        {
+          ClearPreviouslyHitNode();
+          _mode = value;
+        }
+      }
+    }
+
+    public string ToolName => "Edge";
+
+    public bool CanInteractWith(RaycastHit hitInfo)
+    {
+      return _graphService.IsNode(hitInfo.collider.gameObject);
+    }
+
+    public void OnLeftClick(Transform cameraTransform, bool isHit, RaycastHit raycastHit)
+    {
+      Mode = EdgeMode.Create;
+      ModifyEdge(isHit, raycastHit);
+    }
+
+    public void OnRightClick(bool isHit, RaycastHit raycastHit)
+    {
+      Mode = EdgeMode.Delete;
+      ModifyEdge(isHit, raycastHit);
+    }
+
+    public void OnToolsChanged(ITool newTool)
+    {
+      ClearPreviouslyHitNode();
+    }
+
+    private void ClearPreviouslyHitNode()
+    {
+      if (_previouslyHitNode != null)
+      {
+        _previouslyHitNode.gameObject.DisableGlow();
+        _previouslyHitNode = null;
+      }
+    }
+
+    private void ModifyEdge(bool isHit, RaycastHit raycastHit)
+    {
+      if (!isHit)
+      {
+        return;
```

Comment:
You probably won;t agree with me but I hate brackets for a single return command :)